### ISSUE TITLE
Connection refused to MariaDB

### ISSUE THREAD

**Issue by Underlyingglitch** [aschmelyun/docker-compose-laravel#204]
MySQL container is running, but Laravel migrations fail because the connection is refused.

**Comment by kingispeak** [comment]
same problem

**Comment by Underlyingglitch** [comment]
I found the issue. If you are running the MariaDB container for the first time it will need a while to set everything up. During that time it won't accept connections, thus the migrations will fail.

Also make sure you are using a persistent volume for this container, as containers get removed on `docker-compose down`. Mounting a folder with:
```
volumes:
  - ./mysql:/var/lib/mysql
```
makes sure that the files are saved over restarts. Every time you start the stack now it will go as expected.

**Comment by mrbyatga** [comment]
Hello,

I have running mariadb@10.3 in port 3306. So I changed the port '3307:3306' under the mysql in the docker-compose.yml. Also I changed the port to 3307 in the ./src/.env and keep the rest same. When I run the `docker-compose run --rm artisan migrate` I'm getting connection refuse error.

As another approach, I stopped mariadb@10.3 service and using the original configuration of port 3306 from the repository. There is no connection refused error this time and everything went through properly.

Is there any way to use both mariadb@10.3 on port 3306 and mariadb@10.6 from docker on port 3307?

Appreciate and thanks for your reply.

Thanks :)
